packages/matter.js/src/device/PairedNode.ts

```diff
@@ -859,6 +859,11 @@ export class PairedNode {
         const endpointUsages: { [key: EndpointNumber]: EndpointNumber[] } = {};
         Array.from(partLists.entries()).forEach(([parent, partsList]) =>
             partsList.forEach(endPoint => {
+                if (endPoint === parent) {
+                    // There could be more cases of invalid and cycling structures that never should happen ... so lets not over optimize to try to find all of them right now
```

Comment:
Seems worth avoiding all cycles but I suppose we can just make sure we detect when building the ClientNode hierarchy

Reply:
Exactly. I would not spend too much effort now. also because this just happened because of a "custom dev did wrong matter structure".